splunk/komand_splunk/actions/search/action.py

```diff
@@ -2,25 +2,68 @@
 from .schema import SearchInput, SearchOutput, Input, Output, Component
 # Custom imports below
 import json
+from datetime import datetime
+from komand.exceptions import PluginException
 
 
 class Search(komand.Action):
 
     def __init__(self):
         super(self.__class__, self).__init__(
-                name='search',
-                description=Component.DESCRIPTION,
-                input=SearchInput(),
-                output=SearchOutput())
+            name='search',
+            description=Component.DESCRIPTION,
+            input=SearchInput(),
+            output=SearchOutput())
 
     def run(self, params={}):
         """Run action"""
-        result = self.connection.client.jobs.oneshot(
-                params.get(Input.QUERY), count=params.get(Input.COUNT), output_mode="json")
+        search_timeframe = self.parse_search_timeframe(params.get(Input.SEARCH_TIMEFRAME))
+        if search_timeframe:
+            result = self.connection.client.jobs.oneshot(
+                params.get(Input.QUERY),
+                count=params.get(Input.COUNT),
+                output_mode="json",
+                earliest_time=search_timeframe["start_time"],
+                latest_time=search_timeframe["end_time"]
+            )
+        else:
+            result = self.connection.client.jobs.oneshot(
+                params.get(Input.QUERY),
+                count=params.get(Input.COUNT),
+                output_mode="json"
+            )
         results = json.loads(result.readall())
 
         count = 0
         if "results" in results:
             count = len(results["results"])
 
         return {Output.RESULT: results, Output.COUNT: count}
+
+    @staticmethod
+    def parse_search_timeframe(search_timeframe: str) -> dict:
+        datetime_format = "%Y-%m-%dT%H:%M:%S"
+        if search_timeframe:
```

Comment:
This check is being done on line 21 already, can you move this function call into that `if` statement and then remove this line?